Add "Beyond #100000: You're breathtaking!" (#972)
godotengine/godot-website: collections/_article/beyond-100000-you-re-breathtaking.md

---
title: "Beyond #100000: You're breathtaking!"

> excerpt: "As we reach issue #100000 on GitHub, we celebrate with a little retrospective on our issue and PR growth."

Reviewer: ```suggestion
excerpt: "As we reach issue 100,000 on GitHub, we celebrate with a little retrospective on our issue and PR growth."
```

Author: I'm not sure, as it's issue #(number) 100000, isn't it? Maybe #100,000?

Reviewer: Maybe. For some reason, I just thought the # looked out of place in the excerpt, even though it's common syntax in GitHub comments etc. I don't really mind either way though.

categories: ["news"]
author: "Rémi Verschelde"
image: /storage/blog/covers/beyond-100000-you-re-breathtaking.webp
image_caption_title: "\"You're breathtaking!\""
image_caption_description: "Sketch of Keanu Reeves at the 2019 #XboxE3 event"
date: 2024-12-04 19:00:00
---

<div markdown=1 class="card card-info" style="margin-top: 1em;">

> This blog post has been originally posted on [GH-100000](https://github.com/godotengine/godot/issues/100000).

Reviewer: ```suggestion
This blog post was originally posted on [GH-100000](https://github.com/godotengine/godot/issues/100000).
```

All these people brought Juan and Ariel's little-engine-that-could from this:

![Screenshot of the isometric 2D demo in Godot 1.0](/storage/blog/beyond-100000-you-re-breathtaking/godot-1-0.webp)
<div markdown=1 class="small text-center">
*Screenshot of the isometric 2D demo in Godot 1.0.*
</div>

To this:

![Screenshot of PVKK: Planetenverteidigungskanonenkommandant in Godot 4.3](/storage/blog/beyond-100000-you-re-breathtaking/godot-4-3.webp)
<div markdown=1 class="small text-center">
*Screenshot of [PVKK: Planetenverteidigungskanonenkommandant](https://store.steampowered.com/app/2956040/PVKK_Planetenverteidigungskanonenkommandant/) in Godot 4.3.*
</div>

That's no small achievement, so I encourage all contributors to take a minute to contemplate the progress that we've made together over this journey so far!

> Amidst the daily churn of fixing issues, reviewing PRs, making releases, etc., it's important to remind ourselves of where we are, and how we got there.

Reviewer: ```suggestion
Amidst the daily churn of fixing issues, reviewing pull requests, making releases, etc., it's important to remind ourselves of where we are, and how we got there.
```

Author: I wrapped all PR and PRs mentions with \<abbr\>. It looks like this.

<img width="840" alt="Capture d’écran, le 2024-12-04 à 15 29 10" src="https://github.com/user-attachments/assets/46dad9fc-e6a6-41a1-8e1e-2b8718a1adba">

- [**Regularly re-assess your own issues**](https://github.com/godotengine/godot/issues?q=is%3Aissue%20state%3Aopen%20author%3A%40me) (you can bookmark that URL). Make sure that they are reproducible in the latest stable release and dev snapshots, and that there is a minimal reproduction project that contributors can use to reproduce and fix the issue. If your issue is no longer relevant, please close it with a comment explaining why.
- Feel free to apply the same process to other people's issues that you stumble upon, especially if their last update was a long time ago. In this case you can't close issues yourself, but you can suggest it by commenting that the issue is no longer reproducible in the version you tested. Bug triagers will get a notification and can double check.

### Pull Requests

![Graph showing pull request count over time](https://github.com/user-attachments/assets/cf5027a6-4ad3-49bf-9834-d0e1cab2fdc9)
<div markdown=1 class="small text-center">
*Visualization from OSS Insight showing pull request count over time, [see the interactive version](https://ossinsight.io/analyze/godotengine/godot#pr-history).*
</div>

I didn't annotate this one, but a few takeaways:

> - Godot gets almost the same number of new pull requests per month than new issues, roughly 600 of each per month (that's 20 PRs and 20 issues per day).

Reviewer: ```suggestion
- Godot gets almost the same number of new PRs per month as new issues, roughly 600 of each per month (that's 20 PRs and 20 issues per day).
```

- Someone should plot this data with relative percentages, but it seems like the size/complexity of pull requests is slightly increasing recently. Whether that trend gets confirmed or not, I can attest that reviewing 600 PRs per month is hard, and that we need more help for interested contributors and users to both test and review the PRs that get opened on a daily basis.

---

That's all for now, I was already way too long and verbose while drafting this at the last minute to (try to) snatch the 100,000th issue number ;)

Aside from showing some cool numbers, I mostly want to convey that we are well aware that we have a significant backlog, though it's not as dire as it might look from the outside.

To deal with it, we need better triage and review processes (which we are designing now), more volunteers involved in these processes, but also importantly [**more funding**](https://fund.godotengine.org/). Volunteer contributors do a ton of work, but many critical parts of the workflow depend on a few paid contractors, and we need to grow that group to better manage the increasing scale of the project.

> [^1]: I'll make a quick note that 100,000 is the combined number of issues and PRs, which share the same index system on GitHub.<br><br>At the time of writing, we've actually had 53,648 issues and 45,213 PRs created. Astute minds will notice that the sum is not 100,000, the difference comes from spam issues or PRs which have been deleted by GitHub.

Reviewer: ```suggestion
[^1]: I'll make a quick note that 100,000 is the combined number of issues and PRs, which share the same index system on GitHub.<br><br>At the time of writing, we've actually had 53,648 issues and 45,213 PRs created. Astute minds will notice that the sum is not 100,000; the difference comes from spam issues or PRs which have been deleted by GitHub.
```
